File: 028- Elevation Point Finder/ElevationPointFinder.py
def find_elevation_point(arr):
    N = len(arr)
    max_left = float('-inf')
    
    for i in range(N):
        max_left = max(max_left, arr[i])
        is_elevation = True
        for j in range(i + 1, N):
            if arr[j] > arr[i]:
                is_elevation = False
                break
        if max_left <= arr[i] and is_elevation:
            return arr[i]

print(find_elevation_point([1, 2, 3, 4, 3, 2, 1]))  # Output: 4
print(find_elevation_point([5, 3]))                # Output: 5

File: 028- Elevation Point Finder/ElevationPointFinder.cpp
#include<iostream>
#include<vector>
#include<climits>
using namespace std;

int findElevationPoint(vector<int>& arr)
{
    int N = arr.size();
    int maxLeft = INT_MIN;

    for(int i = 0; i < N; i++){
        maxLeft = max(maxLeft, arr[i]);
        bool isElevation = true;
        
        for(int j = i+1; j < N; j++){
            if(arr[j] > arr[i]){
                isElevation = false;
                break;
            }
        }

        if(maxLeft <= arr[i] && isElevation)
        {
        return arr[i];
        }
    }
    return -1;
    
}
int main(){
    vector<int> arr1 = {1,2,3,4,5,3,2,1};
    vector<int> arr2 = {5, 3};

    cout<<findElevationPoint(arr1) << endl;
    cout<<findElevationPoint(arr2) << endl;

    return 0;
}